### PR TITLE
Added Convert syntax errors to violations.

### PR DESCRIPTION
**Pull request by laurentmuller**
This is a implementation for the issue #131.

### CODE REVIEW

**Comment by OwlyCode** [comment]
Hey nice addition!

I'm convinced this should be turned on by default to avoid the tool to completely break on syntax errors like it does at the moment. This also would avoid to introduce a new option to the CLI.

Also, can you add a test for that?

**Comment by laurentmuller** [comment]
OK,

I will turn option on by default and create 2 tests for an invalid template:

- Option on: Must not throw an exception

- Option off: Must throw a `SyntaxError`.

**Comment by laurentmuller** [comment]
I have updated the LintCommand. The optional argument has been changed to `--error`. Please see the [Options with optional arguments](https://symfony.com/doc/current/console/input.html#options-with-optional-arguments). So, if the argument is `false` or missing, the error is not throw.

File: src/Console/LintCommand.php [hunk 1]
```diff
@@ -14,6 +15,7 @@
 use function class_exists;
 use function sprintf;
 
+
```
Comment (oallain) [inline src/Console/LintCommand.php:18]:
```suggestion

```

File: src/Console/LintCommand.php [hunk 2]
```diff
@@ -66,14 +69,28 @@ public function execute(InputInterface $input, OutputInterface $output)
...
+        $validator = $container->get('validator');
 
-            $violations = array_merge($violations, $container->get('validator')->validate(new $ruleset($twigVersion), $tokens));
+        foreach ($files as $file) {
+            $realPath = $file->getRealPath();
+            $source = new Source(
+                file_get_contents($realPath),
+                $realPath,
+                str_replace(realpath($path), rtrim($path, '/'), $realPath)
+                );
```
Comment (oallain) [inline src/Console/LintCommand.php]:
```suggestion
            );
```

File: src/Console/LintCommand.php [hunk 3]
```diff
@@ -124,4 +141,16 @@ private function getSeverityLimit(InputInterface $input): int
...
         }
     }
+
+    /**
+     * Check if empty convert option passed
+     *
+     * @param \Symfony\Component\Console\Input\InputInterface $input
+     * @return bool
+     */
+    private function hasEmptyConvertOption(InputInterface $input)
```
Comment (oallain) [inline src/Console/LintCommand.php]:
```suggestion
    private function hasEmptyConvertOption(InputInterface $input): bool
```

**Comment by laurentmuller** [comment]
Hello,
I can understand why the tests failed. On my computer all the tests pass.
Any idea?

**Comment by laurentmuller** [comment]
tests_twigcs

**Comment by oallain** [comment]
I think it's because php7.0 don't support nullable return.
Error is in this file
https://github.com/friendsoftwig/twigcs/blob/59baa5897d2c92b08221c1e74a311afc5ba018a6/src/TwigPort/SyntaxError.php#L37

We need to increase requirement to PHP 7.1+ ?

File: src/Console/LintCommand.php [hunk 4]
```diff
@@ -29,6 +31,7 @@ public function configure()
             ->addOption('severity', 's', InputOption::VALUE_REQUIRED, 'The maximum allowed error level.', 'warning')
             ->addOption('reporter', 'r', InputOption::VALUE_REQUIRED, 'The reporter to use.', 'console')
             ->addOption('display', 'd', InputOption::VALUE_REQUIRED, 'The violations to display, "' . self::DISPLAY_ALL . '" or "' . self::DISPLAY_BLOCKING . '".', self::DISPLAY_ALL)
+            ->addOption('error', 'e',  InputOption::VALUE_OPTIONAL, 'Throw syntax error when a template contains an invalid token.', false)
```
Comment (Ciloe) [inline src/Console/LintCommand.php]: I think that syntax-error is more explicit

```suggestion
            ->addOption('syntax-error', 'e',  InputOption::VALUE_OPTIONAL, 'Throw syntax error when a template contains an invalid token.', false)
```

File: src/Console/LintCommand.php (hunk 3 above)
Comment (Ciloe) [reply src/Console/LintCommand.php]: This function can be deleted. This is not used

**Review by Ciloe — CHANGES_REQUESTED** [review]

**Comment by laurentmuller** [comment]
I have updated SyntaxError to be compatible with PHP 7.0.

**Comment by oallain** [comment]
> I have updated SyntaxError to be compatible with PHP 7.0.

It fix your issue, but I think it's shame to support already PHP 7.0 ;)

**Review by oallain — APPROVED** [review]

File: src/Console/LintCommand.php (hunk 4 above)
Comment (OwlyCode) [reply src/Console/LintCommand.php]: May I suggest `throw-syntax-errors`? Looks to me that we're ready to merge after that!

**Comment by OwlyCode** [comment]
@laurentmuller I fixed the conflicts and merged, the option has been renamed to `--throw-syntax-errors`

File: src/Console/LintCommand.php [hunk 5]
```diff
@@ -29,6 +31,7 @@ public function configure()
             ->addOption('severity', 's', InputOption::VALUE_REQUIRED, 'The maximum allowed error level.', 'warning')
             ->addOption('reporter', 'r', InputOption::VALUE_REQUIRED, 'The reporter to use.', 'console')
             ->addOption('display', 'd', InputOption::VALUE_REQUIRED, 'The violations to display, "'.self::DISPLAY_ALL.'" or "'.self::DISPLAY_BLOCKING.'".', self::DISPLAY_ALL)
+            ->addOption('throw-syntax-error', 'e',  InputOption::VALUE_OPTIONAL, 'Throw syntax error when a template contains an invalid token.', false)
```
Comment (stof) [inline src/Console/LintCommand.php:34]: Why `VALUE_OPTIONAL` rather than a `VALUE_NONE` flag when the presence of the flag is what matters ?

**Pull request merged**